bor/selection.go

```diff
@@ -65,13 +72,21 @@ func SelectNextProducers(blkHash common.Hash, spanEligibleValidators []hmTypes.V
 
 	// extract seed from hash
 	seedBytes := helper.ToBytes32(blkHash.Bytes()[:32])
-	seed := int64(binary.BigEndian.Uint64(seedBytes[:]))
+
+	seedUint64 := binary.BigEndian.Uint64(seedBytes[:])
+	if seedUint64 > math.MaxInt64 {
+		return nil, fmt.Errorf("seed value out of range for int64: %d", seedUint64)
+	}
+	seed := int64(seedUint64)
 	// nolint: staticcheck
 	rand.Seed(seed)
 
 	// weighted range from validators' voting power
 	votingPower := make([]uint64, len(spanEligibleValidators))
 	for idx, validator := range spanEligibleValidators {
+		if validator.VotingPower < 0 {
```

Comment:
Are we sure that it is safer to return error than not? I see we dont have such case but still maybe just skip the -1 and log error?